packages/node/src/evm/verification/request-verification.ts

```diff
@@ -46,14 +41,78 @@ export function verifyDesignatedWallets<T>(
       return [[log], updatedRequest];
     }
 
-    const log = logger.pend(
-      'DEBUG',
-      `Request ID:${request.id} is linked to a valid designated wallet:${request.designatedWallet}`
-    );
+    const message = `Request ID:${request.id} is linked to a valid designated wallet:${request.designatedWallet}`;
+    const log = logger.pend('DEBUG', message);
     return [[log], request];
   });
 
   const logs = flatMap(logsWithVerifiedRequests, (r) => r[0]);
   const verifiedRequests = flatMap(logsWithVerifiedRequests, (r) => r[1]);
   return [logs, verifiedRequests];
 }
+
+export function verifyTriggers(
+  apiCalls: ClientRequest<ApiCall>[],
+  triggers: RequestTrigger[],
+  oises: OIS[]
+): LogsData<ClientRequest<ApiCall>[]> {
+  const logsWithVerifiedApiCalls: LogsData<ClientRequest<ApiCall>>[] = apiCalls.map((apiCall) => {
+    if (apiCall.status !== RequestStatus.Pending) {
+      const message = `Trigger verification skipped for Request:${apiCall.id} as it has status:${apiCall.status}`;
+      const log = logger.pend('DEBUG', message);
+      return [[log], apiCall];
+    }
+
+    const trigger = triggers.find((t) => t.endpointId === apiCall.endpointId);
+    // If the request is for an unknown endpointId, the problem is with the requesting client contract
+    if (!trigger) {
+      const message = `Request:${apiCall.id} has no matching endpointId:${apiCall.endpointId} in Airnode config`;
+      const log = logger.pend('WARN', message);
+      const updatedApiCall = {
+        ...apiCall,
+        status: RequestStatus.Errored,
+        errorCode: RequestErrorCode.UnknownEndpoint,
+      };
+      return [[log], updatedApiCall];
+    }
+
+    const { oisTitle, endpointName } = trigger;
+    const ois = oises.find((o) => o.title === oisTitle);
+    // Although unlikely, it is possible that the Airnode instance was configured with a trigger
+    // with an 'oisTitle' that does not match an OIS in config.json. The responsibility to fix this
+    // lies with the API provider
+    if (!ois) {
+      const log = logger.pend('ERROR', `Unknown OIS:${oisTitle} received for Request:${apiCall.id}`);
+      const updatedApiCall = {
+        ...apiCall,
+        status: RequestStatus.Errored,
+        errorCode: RequestErrorCode.UnknownOIS,
```

Comment:
and these two can be `UnknownEndpoint`

Reply:
This can also be `UnknownOIS` but I see the point of not wanting to have too many error codes

Reply:
I think a single error code should be fine to indicate to the API provider that something is wrong with their config and that they need to look into it